shell/server/mail.js

```diff
@@ -535,6 +538,256 @@ HackSessionContextImpl.prototype.revokeApiToken = function (tokenId) {
   }).bind(this));
 }
 
+function IpInterfaceImpl () { }
+
+IpInterfaceImpl.prototype.listenTcp = function (portNum, port) {
+  return new Promise(function (resolve, reject) {
+    var resolved = false;
+    var server = Net.createServer(function (connection) {
+      connection.done = connection.end;
```

Comment:
This is a little too clever for me. :)

Can we explicitly define an object here meant to implement the Cap'n Proto interface?

I worry about things like we add a new method to ByteStream which happens to conflict with a method in node's connection type, and then suddenly things inexplicably don't work.